source/Projectile.cpp

```diff
@@ -52,7 +52,8 @@ Projectile::Projectile(const Ship &parent, Point position, Angle angle, const We
 	if(inaccuracy)
 		this->angle += Angle::Random(inaccuracy) - Angle::Random(inaccuracy);
 
-	velocity += this->angle.Unit() * (weapon->Velocity() + Random::Real() * weapon->RandomVelocity());
+	marginalVelocity = weapon->Velocity() + Random::Real() * weapon->RandomVelocity();
+	velocity += this->angle.Unit() * marginalVelocity;
```

Comment:
If we continue to use a variable here to contain the new speed of the projectile, these lines could remain almost the same, with `marginalVelocity` becoming a local variable and a 3rd line updating the value of the 'totalProjectileVelocity' class attribute.